packages/caliper-fabric/test/connector-versions/peer-gateway/PeerGateway.js

```diff
@@ -0,0 +1,498 @@
+        Gateway.connectArgs[0].evaluateOptions().should.be.deep.equal({deadline: now.getTime() + 60000});
+        clock.restore();
+    });
+
+    it('should pass the timeout from invokeOrQuery in the gateway options', async () => {
+        ConfigUtil.set(ConfigUtil.keys.Fabric.Timeout.InvokeOrQuery, 99);
+        const connectorConfiguration = await new ConnectorConfigurationFactory().create(path.resolve(__dirname, configWith2Orgs1AdminInWalletNotMutual), walletFacadeFactory);
+        const peerGateway = new PeerGateway(connectorConfiguration, 1, 'fabric');
+        const now = new Date();
+        const clock = sinon.useFakeTimers(now.getTime());
+        await peerGateway.getContext();
+        Gateway.connectArgs[0].evaluateOptions().should.be.deep.equal({deadline: now.getTime() + 99000});
+        clock.restore();
+        ConfigUtil.set(ConfigUtil.keys.Fabric.Timeout.InvokeOrQuery, null);
+    });
+
+    it('should create one Client for each organization', async () => {
+        const connectorConfiguration = await new ConnectorConfigurationFactory().create(path.resolve(__dirname, configWith2Orgs1AdminInWalletNotMutual), walletFacadeFactory);
+        const peerGateway = new PeerGateway(connectorConfiguration, 1, 'fabric');
+        await peerGateway.getContext();
+        GrpcClient.constructed.should.equal(2);
+        GrpcClient.options[0]['grpc.hostnameOverride'].should.deep.equal('peer0.org1.example.com');
+        GrpcClient.options[1]['grpc.hostnameOverride'].should.deep.equal('peer0.org2.example.com');
+    });
+
+    it('should create one Gateway for each identity', async () => {
+        const connectorConfiguration = await new ConnectorConfigurationFactory().create(path.resolve(__dirname, configWith2Orgs1AdminInWalletNotMutual), walletFacadeFactory);
+        const peerGateway = new PeerGateway(connectorConfiguration, 1, 'fabric');
+        await peerGateway.getContext();
+        Gateway.constructed.should.equal(4);
+    });
+
+    it('should a attempt to create a grpc client with default grpc options when not provided through the connection profile', async () => {
+        const connectorConfiguration = await new ConnectorConfigurationFactory().create(path.resolve(__dirname, configWith2Orgs1AdminInWalletNotMutual), walletFacadeFactory);
+        const peerGateway = new PeerGateway(connectorConfiguration, 1, 'fabric');
+        await peerGateway.getContext();
+        GrpcClient.options[0].should.deep.equal({
+            'grpc.http2.max_pings_without_data': 0 ,
+            'grpc.http2.min_time_between_pings_ms': 120000 ,
+            'grpc.keepalive_permit_without_calls': 1 ,
+            'grpc.keepalive_timeout_ms': 20000 ,
+            'grpc.keepalive_time_ms': 120000,
+            'grpc.max_receive_message_length' : -1,
+            'grpc.max_send_message_length' : -1,
+            'grpc.ssl_target_name_override' : 'peer0.org1.example.com',
+            'grpc.hostnameOverride' : 'peer0.org1.example.com'
```

Comment:
rename 'grpc.hostnameOverride' to 'hostnameOverride' we can expect this to be in a connection profile from test network and so it will be in the options, but it will be ignored so we don't need to worry about it. Alternatively you could take hostnameOverride out of the sample connection profiles but I don't think that is necessary